### PR TITLE
Odd fixes and other bits preparing for main menu implemention.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by OmniBlade** [comment]
You need to do an interactive rebase of this branch to rebased it on top of develop and drop the baseconfig update commit or at least just drop the baseconfig update all together.

File: src/game/ui/textbtn.cpp [hunk 1]
```diff
@@ -124,30 +100,26 @@ void TextButtonClass::Set_Text(int str_id, BOOL adjust)
...
-
 void TextButtonClass::Draw_Background()
 {
     BoxStyleEnum style;
 
     if (HasOutline) {
         g_logicPage->Draw_Rect(XPos - 1, YPos - 1, Width + XPos + 2, Height + YPos + 2, COLOR_BLACK);
+    }
 
-        // Draw an drop shadow underneath the button if required.
```
Comment (OmniBlade) [inline src/game/ui/textbtn.cpp]: The style change for the comments is somewhat pointless, I'd recommend just dropping the commented out code all together unless/until we want to actually implement it as its in the commit history if we want to pull it back.

File: src/game/engine/target.cpp [hunk 2]
```diff
@@ -101,6 +101,11 @@ target_t As_Target(CellClass *cell)
     return cell != nullptr ? As_Target(cell->Cell_Number()) : 0;
 }
 
+target_t As_Target(AbstractClass *object)
```
Comment (OmniBlade) [inline src/game/engine/target.cpp]: This isn't required, I already added a method to AbstractClass to return itself as a target_t.

File: src/game/engine/version.cpp [hunk 3]
```diff
@@ -17,11 +17,9 @@
 #include "expansion.h"
 #include "gitverinfo.h"
 #include "rawfile.h"
-#include <cstdio>
-#include <cstring>
-
-using std::strlen;
-using std::snprintf;
+#include "language.h"
```
Comment (OmniBlade) [inline src/game/engine/version.cpp]: Not sure why these header changes are here.

File: src/game/engine/display.h [hunk 4]
```diff
@@ -81,6 +81,7 @@ class DisplayClass : public MapClass
 
     coord_t Pixel_To_Coord(int x, int y);
     void Set_Cursor_Shape(int16_t *list = nullptr);
+    void Clear_Cursor_Shape() { Set_Cursor_Shape(nullptr); }
```
Comment (OmniBlade) [inline src/game/engine/display.h]: What does this do that calling `Set_Cursor_Shape();` doesn't do? Seems pointless, its even slightly longer to type.

File: src/game/common/globals.h [hunk 5]
```diff
@@ -26,6 +26,57 @@
 #define REDRAW_GADGETS 1
 #define REDRAW_ALL 2
 
+//
```
Comment (OmniBlade) [inline src/game/common/globals.h]: Is globals.h really the right place for all these gadget values?

File: src/game/engine/display.h (hunk 4 above)
Comment (OmniBlade) [inline src/game/engine/display.h]: What does this do that calling Set_Cursor_Shape(); doesn't do? Seems pointless, its even slightly longer to type.

File: src/game/engine/version.cpp [hunk 6]
```diff
@@ -17,11 +17,9 @@
...
+#include "stringex.h"
```
Comment (OmniBlade) [inline src/game/engine/version.cpp]: Not sure why these header changes are here.

File: src/game/engine/target.cpp (hunk 2 above)
Comment (OmniBlade) [inline src/game/engine/target.cpp:104]: This isn't required, I already added a method to AbstractClass to return itself as a target_t.

File: src/game/ui/textbtn.cpp (hunk 1 above)
Comment (OmniBlade) [inline src/game/ui/textbtn.cpp]: The style change for the comments is somewhat pointless, I'd recommend just dropping the commented out code all together unless/until we want to actually implement it as its in the commit history if we want to pull it back.

**Review by OmniBlade — CHANGES_REQUESTED** [review]

File: src/game/engine/display.h (hunk 4 above)
Comment (MrSapps) [reply src/game/engine/display.h]: You could just default the argument to Set_Cursor_Shape to be nullptr instead anyway

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**